DatadogSessionReplay/Sources/DatadogCoreIntegration/SRContextPublisher.swift

```diff
@@ -22,4 +22,12 @@ internal class SRContextPublisher {
             attributes: { [RUMDependency.hasReplay: value] }
         )
     }
+
+    /// Notifies other Features on the state of  Session Replay recording.
+    func setRecordsCount(_ value: [String: Int64]) {
+        core?.set(
+            feature: RUMDependency.srBaggageKey,
+            attributes: { [RUMDependency.recordsCount: value] }
+        )
```

Comment:
blocker/ This resets the previously set context (including `has_replay` value ⚠️). Can be noticed if adding more tests to `SRContextPublisher` (below).

<img width="1362" alt="Screenshot 2023-07-04 at 11 01 37" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/234bd834-e332-4859-918f-135b5a953fc8">